R/SingleR.visualizations.R

```diff
@@ -0,0 +1,121 @@
+#' Plot a scatter plot of a single cell vs. a single reference cell/sample
+#'
+#' @param sc_data Similar to SinlgeR test input, a numeric matrix of single-cell expression values (usually
+#' log-transformed or otherwise variance-stabilized), where rows are genes and columns are cells.
+#' Alternatively, a \linkS4class{SingleCellExperiment} object containing such a matrix.
+#' @param cell_id a number indicating which single cell to use.
+#' Alternatively, the name of the cell.
+#' @param refdata Similar to SinlgeR training input, a numeric matrix of expression values. 
+#' Alternatively, a \linkS4class{SingleCellExperiment} object containing such a matrix.
+#' This should have the same rows as or a subset of the rows in \code{test}.
+#' @param sample_id a number of the sample to use
+#' @param assay.type An integer scalar or string specifying the assay of \code{sc_data} or \code{refdata} containing the relevant expression matrix,
+#' if \code{sc_data} or \code{refdata} is a \linkS4class{SingleCellExperiment} object.
+#'
+#' @return a gpplot scatterplot of the expression of indidivual genes, with linear regression and Spearman pariwaise correlation coefficient
+#' 
+#' @export
+#' @importFrom ggplot2 ggplot geom_point geom_smooth theme xlab ylab ggtitle theme_classic
+SingleR.DrawScatter = function(sc_data, cell_id, refdata,sample_id, assay.type = 1) {
```

Comment:
The `SingleR.` is unnecessary, as people can just do `SingleR::` to get all functions associated with a package. I would suggest naming as just `drawScatter()` - or possibly with a more meaningful name about what the function actually does, e.g., `plotCellToReference()`. 

Also some more notes about Bioconductor style:

- Upper case first letter is reserved for class names (e.g., `SummarizedExperiment`) and their constructors. Functions generally have a lower case first letter. Unfortunately, we're kind of stuck with `SingleR()` as this is the package name, but we should try to be consistent everywhere else. 
- Consider synchronizing the argument names here to the names in the other functions, e.g., in `SingleR()`. This should make things easier to digest. We should pick a standard set of names to use in all functions - if you come up with one, feel free to implement it. (I would prefer `.` instead of `_` for separators, though.)

Reply:
I went with plotCellVsReference().
Also, I have ideas for making all the names similar... I'm putting my suggested input names in these functions.  I will make my suggestions in the main functions after.

Reply:
Yes, we should discuss good names before either of us does anything package-wide.